client/log_client_test.go

```diff
@@ -54,12 +55,18 @@ func TestGetByIndex(t *testing.T) {
 		t.Fatal(err)
 	}
 	defer env.Close()
-	logID, err := env.CreateLog()
+
+	tree, err := CreateAndInitTree(ctx, &trillian.CreateTreeRequest{
+		Tree: stestonly.LogTree,
+	}, env.Admin, nil, env.Log)
 	if err != nil {
 		t.Fatalf("Failed to create log: %v", err)
 	}
 
-	client := New(logID, env.Log, rfc6962.DefaultHasher, env.PublicKey)
+	client, err := NewFromTree(env.Log, tree)
```

Comment:
I can't easily tell where `rfc6962.DefaultHasher, env.PublicKey` ended up after your refactor in this PR?

Reply:
Good question.  They are inside the `*trillian.Tree` object which is returned from `CreateAndInitTree`. 

The hasher and public key are unique to each tree rather than the server.

Reply:
OK, makes sense.